src/Microsoft.OpenApi.OData.Reader/Operation/DollarCountGetOperationHandler.cs

```diff
@@ -75,19 +88,85 @@ protected override void SetResponses(OpenApiOperation operation)
             base.SetResponses(operation);
         }
 
-        protected override void AppendCustomParameters(OpenApiOperation operation)
+        /// <inheritdoc/>
+        protected override void SetParameters(OpenApiOperation operation)
         {
-            IEdmVocabularyAnnotatable annotatable = null;            
-            if (LastSecondSegment is ODataNavigationSourceSegment sourceSegment)
+            base.SetParameters(operation);
+            if (annotatable is IEdmSingleton singleton)
             {
-                annotatable = sourceSegment.NavigationSource as IEdmEntitySet;
-                annotatable ??= sourceSegment.NavigationSource as IEdmSingleton;
+                // $select
+                OpenApiParameter parameter = Context.CreateSelect(singleton);
+                if (parameter != null)
+                {
+                    operation.Parameters.Add(parameter);
+                }
+
+                // $expand
+                parameter = Context.CreateExpand(singleton);
+                if (parameter != null)
+                {
+                    operation.Parameters.Add(parameter);
+                }
             }
-            else if (LastSecondSegment is ODataNavigationPropertySegment navigationPropertySegment)
+            else
             {
-                annotatable = navigationPropertySegment.NavigationProperty;
+                // EntitySet or Collection-Valued Navigation Property
+
+                OpenApiParameter parameter = Context.CreateTop(annotatable);
```

Comment:
> The /$count path suffix identifies the integer count of records in the collection and SHOULD NOT be combined with the system query options  [$top](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionstopandskip), [$skip](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionstopandskip), [$orderby](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionorderby), [$expand](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionexpand), and [$format](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionformat). The count MUST NOT be affected by $top, $skip, $orderby, or $expand.

Please remove skip, top, search, expand, count, select, orderby

Reply:
That means we only create `$filter` and `$search` and only if the parent endpoint is a collection-valued nav. prop. or an entity set?

Reply:
correct, I believe the raw count path segment will only ever get added after those anyway with the code we have today.